File: 1351.cpp
/*
	Author: Zoljargal Gantumur
	Runtime: 24 ms, faster than 88.85% of C++ online submissions
	Memory Usage: 10.6 MB, less than 54.54% of C++ online submissions
*/ 

class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        std::ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        int count=0;
        int m=grid.size();
        int n=grid[0].size();
        for(int i=0;i<m;i++)
        {
            for(int j=n-1;j>=0;j--)
            {
                if(grid[i][j]>=0)
                    break;
                 count++;
            }
        }
        return count;
    }
};

File: 1351.py
# Author: Zoljargal Gantumur
# Runtime: 112 ms, faster than 99.59% of Python3 online submissions
# Memory Usage: 14.5 MB, less than 95.10% of Python3 online submissions

class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        count = 0
        m = len(grid)
        n = len(grid[0])
        for i in range(m):
            low = 0 
            high = n-1
            res = -1
            while low <= high:
                mid = (low+high)//2
                if grid[i][mid] >= 0:
                    low = mid + 1
                elif grid[i][mid] < 0:
                    res = mid
                    high = mid - 1
            if res!=-1:
                count += (n-res)
        return count
